### PR TITLE
AMQ-7159 - Adding a new attribute on PersistenceAdapterViewMBean to s…

### PR DESCRIPTION
**Pull request by alanprot**
…how information about Storage write latency

### CODE REVIEW

File: activemq-broker/src/main/java/org/apache/activemq/broker/jmx/PersistenceAdapterView.java [hunk 1]
```diff
@@ -64,11 +88,46 @@ private String invoke(Callable<String> callable) {
         return result;
     }
 
+    private String serializePersistenceAdapterStatistics() {
+        if (persistenceAdapterStatistics != null) {
+            try {
+                final ObjectMapper mapper = new ObjectMapper();
```
Comment (cshannon) [inline activemq-broker/src/main/java/org/apache/activemq/broker/jmx/PersistenceAdapterView.java]: ObjectMapper is threadsafe so you should just create it one time for the class so we don't need to re-create it over and over.

File: activemq-broker/src/main/java/org/apache/activemq/store/PersistenceAdapterStatistics.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package org.apache.activemq.store;
+
+import org.apache.activemq.management.TimeStatisticImpl;
+
+public class PersistenceAdapterStatistics {
```
Comment (cshannon) [inline activemq-broker/src/main/java/org/apache/activemq/store/PersistenceAdapterStatistics.java]: This looks fine except I think the rest of the statistics classes extend StatsImpl so you may want to follow that pattern to be consistent.  (For example look at DestinationStatistics)

File: activemq-kahadb-store/src/main/java/org/apache/activemq/store/kahadb/MessageDatabase.java [hunk 3]
```diff
@@ -249,6 +250,7 @@ public void writePayload(Metadata object, DataOutput dataOut) throws IOException
     protected PageFile pageFile;
     protected Journal journal;
     protected Metadata metadata = new Metadata();
+    protected PersistenceAdapterStatistics persistenceAdapterStatistics = new PersistenceAdapterStatistics();
```
Comment (cshannon) [inline activemq-kahadb-store/src/main/java/org/apache/activemq/store/kahadb/MessageDatabase.java]: I would probably make this final and then I would also add a getter for this object and also expose it under KahaDBPersistenceAdapter. I routinely use embedded brokers for example and being able to access this through KahaDBPersistenceAdapter without JMX would be nice.  Inside KahaDBPersistenceAdapter I would just return the actual PersistenceAdapterStatistics object so someone could access it.

File: activemq-broker/src/main/java/org/apache/activemq/broker/jmx/PersistenceAdapterView.java (hunk 1 above)
Comment (alanprot) [reply activemq-broker/src/main/java/org/apache/activemq/broker/jmx/PersistenceAdapterView.java]: Good catch!

File: activemq-broker/src/main/java/org/apache/activemq/store/PersistenceAdapterStatistics.java (hunk 2 above)
Comment (alanprot) [reply activemq-broker/src/main/java/org/apache/activemq/store/PersistenceAdapterStatistics.java]: Done! :)

File: activemq-kahadb-store/src/main/java/org/apache/activemq/store/kahadb/MessageDatabase.java (hunk 3 above)
Comment (alanprot) [reply activemq-kahadb-store/src/main/java/org/apache/activemq/store/kahadb/MessageDatabase.java]: Done.

**Pull request merged**